lahja/tools/benchmark/process.py

```diff
@@ -56,79 +61,85 @@ def start(self) -> None:
 
     @staticmethod
     def launch(config: DriverProcessConfig) -> None:
-        loop = asyncio.get_event_loop()
-        event_bus = Endpoint()
-        event_bus.start_serving_nowait(ConnectionConfig.from_name(DRIVER_ENDPOINT))
-        event_bus.connect_to_endpoints_blocking(*config.connected_endpoints)
         # UNCOMMENT FOR DEBUGGING
         # logger = multiprocessing.log_to_stderr()
         # logger.setLevel(logging.INFO)
-        loop.run_until_complete(DriverProcess.worker(event_bus, config))
 
-    @staticmethod
-    async def worker(event_bus: Endpoint, config: DriverProcessConfig) -> None:
-        payload = b'\x00' * config.payload_bytes
-        for n in range(config.num_events):
-            await asyncio.sleep(config.throttle)
-            event_bus.broadcast(
-                PerfMeasureEvent(payload, n, time.time())
-            )
+        # wait until the reporter and all the consumers are ready
+        for event in config.ready_events:
+            event.wait()
 
-        event_bus.stop()
+        loop = asyncio.get_event_loop()
+        loop.run_until_complete(DriverProcess.worker(config))
+
+    @staticmethod
+    async def worker(config: DriverProcessConfig) -> None:
+        with Endpoint() as event_bus:
+            await event_bus.start_serving(ConnectionConfig.from_name(DRIVER_ENDPOINT))
+            await event_bus.connect_to_endpoints(*config.connected_endpoints)
+
+            payload = b'\x00' * config.payload_bytes
```

Comment:
Not needed for this PR but it might be good to try and use representative data.

- text from a book that's in the creative commons
- data from the [`faker`](https://faker.readthedocs.io/en/master/providers.html)

Reply:
https://github.com/ethereum/lahja/issues/40

Reply:
Yeah, definitely! This is a reason why at one point benchmarking would hang (on the massive messages) but snappy-benchmark worked, the snappy messages compress down to nothing :smile: